lib/dataset/test/dataset_arithmetic_test.cpp

```diff
@@ -107,6 +115,10 @@ TYPED_TEST(DataArrayViewBinaryEqualsOpTest, other_data_unchanged) {
     const auto original_a = copy(dataset_a);
     auto target = dataset_a["data_zyx"];
 
+    if (item.has_variances() && target.dims() != item.dims()) {
+      ASSERT_THROW(TestFixture::op(target, item), except::VariancesError);
+      continue;
```

Comment:
It's not great to have an `if` here. But I guess the test is pretty complicated anyway. It would needs a large refactor to eliminate the `if`.

But is checking variance handling the purpose of this test? Or should the input simple not have variances at all?

Reply:
Changing to use `make_no_variances`.